gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/orc/OrcKeyDedupReducer.java

```diff
@@ -37,6 +53,30 @@ protected void setOutKey(OrcValue valueToRetain) {
     // do nothing since initReusableObject has assigned value for outKey.
   }
 
+  @Override
+  protected void reduce(OrcKey key, Iterable<OrcValue> values, Context context)
+      throws IOException, InterruptedException {
+
+    /* Map from hash of value(Typed in OrcStruct) object to its times of duplication*/
+    Map<Integer, Integer> valuesToRetain = new HashMap<>();
+    int valueHash = 0;
+
+    for (OrcValue value : values) {
```

Comment:
The benefit is during the sorting phase and MR is going to compare key, the comparison is down to a much shortened object instead of the whole object (and we get rid of complex column like array , map, etc.).  This is improving map throughput quite a bit, and on the reducer side we are not doing much more if the shuffle key is selected properly: We chose uuid + header.time and chances for two different records to collide on this pair is pretty low.

Reply:
Thanks for the explanation. Makes sense.